lightning/src/blinded_path/message.rs

```diff
@@ -85,37 +86,73 @@ impl Writeable for ReceiveTlvs {
-/// if needed.
+/// This data is encrypted by the recipient and will be given to the corresponding message handler
+/// when handling a message sent over the [`BlindedPath`]. The recipient can use this data to
+/// authenticate the message or for further processing if needed.
 #[derive(Clone, Debug)]
 pub enum MessageContext {
-	/// Represents the data specific to [`OffersMessage`]
+	/// Context specific to an [`OffersMessage`].
 	///
 	/// [`OffersMessage`]: crate::onion_message::offers::OffersMessage
 	Offers(OffersContext),
-	/// Represents custom data received in a Custom Onion Message.
+	/// Context specific to a [`CustomOnionMessageHandler::CustomMessage`].
+	///
+	/// [`CustomOnionMessageHandler::CustomMessage`]: crate::onion_message::messenger::CustomOnionMessageHandler::CustomMessage
 	Custom(Vec<u8>),
 }
 
 /// Contains the data specific to [`OffersMessage`]
 ///
 /// [`OffersMessage`]: crate::onion_message::offers::OffersMessage
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Eq, PartialEq)]
 pub enum OffersContext {
 	/// Represents an unknown BOLT12 payment context.
-	/// This variant is used when a message is sent without
-	/// using a [`BlindedPath`] or over one created prior to
-	/// LDK version 0.0.124.
+	///
+	/// This variant is used when a message is sent without using a [`BlindedPath`] or over one
+	/// created prior to LDK version 0.0.124.
 	Unknown {},
-	/// Represents an outbound BOLT12 payment context.
+	/// Context used by a [`BlindedPath`] within an [`Offer`] for sending an [`InvoiceRequest`].
+	///
+	/// [`Offer`]: crate::offers::offer::Offer
+	/// [`InvoiceRequest`]: crate::offers::invoice_request::InvoiceRequest
+	InvoiceRequest {
+		/// A nonce used for authenticating that an [`InvoiceRequest`] is for a valid [`Offer`] and
+		/// for deriving the offer's signing keys.
+		///
+		/// [`InvoiceRequest`]: crate::offers::invoice_request::InvoiceRequest
+		/// [`Offer`]: crate::offers::offer::Offer
+		nonce: Nonce,
+	},
+	/// Context used by a [`BlindedPath`] within a [`Refund`] or as a reply path for an
+	/// [`InvoiceRequest`].
+	///
+	/// [`Refund`]: crate::offers::refund::Refund
+	/// [`InvoiceRequest`]: crate::offers::invoice_request::InvoiceRequest
 	OutboundPayment {
-		/// Payment ID of the outbound BOLT12 payment.
-		payment_id: PaymentId
+		/// Payment ID used when creating a [`Refund`] or [`InvoiceRequest`].
+		///
+		/// [`Refund`]: crate::offers::refund::Refund
+		/// [`InvoiceRequest`]: crate::offers::invoice_request::InvoiceRequest
+		payment_id: PaymentId,
+		/// A nonce used for authenticating that an [`Invoice`] is for a valid [`Refund`] or
+		/// [`InvoiceRequest`] and for deriving their signing keys.
+		///
+		/// [`Refund`]: crate::offers::refund::Refund
+		/// [`InvoiceRequest`]: crate::offers::invoice_request::InvoiceRequest
+		nonce: Nonce,
+	},
+	/// Context used by a [`BlindedPath`] for replying to a [`Bolt12Invoice`] with an
+	/// [`InvoiceError`].
```

Comment:
nit: I think we can reword it a bit for better clarity!

```suggestion
	/// Context used by a [`BlindedPath`] as a reply path for a [`Bolt12Invoice`].
```

Reply:
Also updated this to align with the other variants, for consistency.